### ISSUE TITLE
Add API to build target transforms for datasets

### ISSUE THREAD

**Issue by satyajitghana** [extensive-vision-ai/TheTensorClan#2]
for target transforms maybe we can have like `build_target_transforms(train=False|True)` ?

What the current API is:

```python
class BaseAugmentation(metaclass=ABCMeta):
    def build_transforms(self, train: bool):
        if train:
            return AlbumentationTransforms(self.build_train())
        else:
            return AlbumentationTransforms(self.build_test())

    @abstractmethod
    def build_train(self):
        pass

    @abstractmethod
    def build_test(self):
        pass
```

**Comment by hemanth346** [comment]
Can you elaborate further.? 
My understanding is that target is the image we send as input, and the transformations is already being taken care of by build_transfroms, what are the expectations of the _build_target_transforms_?

**Comment by satyajitghana** [comment]
> Can you elaborate further.?
> My understanding is that target is the image we send as input, and the transformations is already being taken care of by build_transfroms, what are the expectations of the _build_target_transforms_?

by build_target_transforms i mean building albumentations transforms of the label/target of the dataset
build_transforms -> for transforming the input_image/text or anything
build_target_transforms -> for transforming the label of the respective input

**Comment by satyajitghana** [comment]
i really don't like the way i do, build_x_transforms and having a wrapper class for albumentations, can we have some alternative, more cleaner way ?
the only reason i have a class wrapper and build method for transforms is because albumentations doesnt work like `torchvision.transforms`, but i was thinking, we could override `__call__` method to avoid that wrapper class somehow ?

**Comment by hemanth346** [comment]
I still didn't get understand, Can you maybe write a class for a particular dataset or maybe refer to any api/code where you came across the need? I'm not sure why do we have to transform labels.
For integrating albumentations, I'll check on that

**Comment by satyajitghana** [comment]
take a look at torchvision datasets https://github.com/pytorch/vision/blob/master/torchvision/datasets/vision.py , they have this parameter for target transforms, for a use case that i can think of would be like lets say there is a dataset in which the target is a image, something like our depth dataset in phase-1, we can have a target_transform to transform the target image.
